sktime/forecasting/boxcox_bias_adjusted_forecaster.py

```diff
@@ -0,0 +1,202 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+"""BoxCoxBiasAdjustedForecaster implementation."""
+
+__author__ = ["sanskarmodi8"]
+
+import pandas as pd
+from scipy.special import inv_boxcox
+
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.base._delegate import _DelegatedForecaster
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class BoxCoxBiasAdjustedForecaster(BaseForecaster):
+    r"""Box-Cox Bias-Adjusted Forecaster.
+
+    This module implements a forecaster that applies Box-Cox transformation
+    and bias adjustment to the predictions of a wrapped forecaster.
+
+    The bias adjustment is applied to both point predictions and prediction
+    intervals using the bias correction formula for inverse Box-Cox transforms:
+
+    .. math::
+
+        y_t = \text{inv\_boxcox}(w_t, \lambda) \cdot [1 + \sigma_t^2(1-\lambda)/
+        (2(\lambda w_t + 1)^2)]
+
+    where:
+    - :math:`y_t` is the back-transformed forecast at time t
+    - :math:`w_t` is the transformed data at time t
+    - :math:`\lambda` is the Box-Cox parameter
+    - :math:`\sigma_t^2` is the forecast variance on the transformed scale at time t
+
+    Parameters
+    ----------
+    forecaster : BaseForecaster
+        The wrapped forecaster to which Box-Cox transformation and
+        bias adjustment will be applied.
+    lambda_fixed : float, optional (default=None)
+        The Box-Cox transformation parameter. If None, it will be estimated.
+
+    References
+    ----------
+    .. [1] Hyndman, R.J., & Athanasopoulos, G. (2018) "Forecasting:
+           principles and practice", 2nd edition, Section 2.7 - Box-Cox Transformations,
+           OTexts: Melbourne, Australia. OTexts.com/fpp2
+    """
+
+    _tags = {
+        "capability:pred_int": True,
+        "capability:pred_var": True,
+    }
+
+    def __init__(self, forecaster, lambda_fixed=None):
+        self.forecaster = forecaster
+        self.lambda_fixed = lambda_fixed
+        self.boxcox_transformer_ = None
+        self._y_name = None
+        _DelegatedForecaster._set_delegated_tags(self, forecaster)
+        super().__init__()
+
+    def _get_delegate(self):
+        """Return the delegate forecaster."""
+        return self.forecaster
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the forecaster to the training data.
+
+        Parameters
+        ----------
+        y : pd.Series or pd.DataFrame
+            Target time series to which to fit the forecaster.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables.
+        fh : ForecastingHorizon
+            The forecasting horizon with the steps ahead to predict.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        self._y_name = y.name if isinstance(y, pd.Series) else None
```

Comment:
where did the "if not get_tag pred_int" go? That was a reasonable line to check.

Reply:
Actually even without that line we were getting appropiate error message regarding the same if that tag in the forecaster is set to `False`. Should I reimplement the check?

Reply:
Yes, because I think the default message does not explain to the user why `predict_var` gets called. The error message should explain that the new forecaster requires a `forecaster` with the capability.